RecordCollection/Controllers/AlbumsController.cs

```diff
@@ -15,8 +19,39 @@ public AlbumsController(RecordCollectionContext context)
         public IActionResult Index()
         {
             var albums = _context.Albums.ToList();
-            
             return View(albums);
         }
+
+        [Route("Albums/{id:int}")]
+        public async Task<IActionResult> Show(int? id)
+        {
+            if (id == null)
+            {
+                return NotFound();
+            }
+
+            var retrieved = await _context.Albums.FindAsync(id);
+
+            var relatedAlbums = _context.Albums
+                .Where(a => a.Artist == retrieved.Artist && a.Id != retrieved.Id)
+                .OrderBy(a => a.ReleaseDate);
+
+            var viewModel = new AlbumViewModel
+            {
+                Album = retrieved,
+                RelatedAlbums = relatedAlbums.ToList()
+            };
+
+            if (viewModel.RelatedAlbums.Count > 0)
+            {
+                viewModel.RelatedAlbumsCount = viewModel.RelatedAlbums.Count;
+            }
+            else
+            {
+                viewModel.RelatedAlbumsCount = 0;
+            }
+
```

Comment:
We can use the Move method refactoring pattern on lines 45-52 and move this functionality into the AlbumViewModel class itself, as it only uses and sets information from that class